app/Dockerfile

```diff
@@ -14,11 +14,11 @@ RUN npm install --quiet
 # Bundle app source
 COPY . /usr/src/app
 
-ENV NODE_ENV=production
+ENV STAGING=production
 ENV PORT=4000
 
 RUN ./node_modules/.bin/webpack --config webpack.config.js
 
-EXPOSE 3000
+EXPOSE 4000
```

Comment:
Why 4000 instead of 3000 (although I don't care, it's just to know why you choose to change it, like if there was a conflict or so)